src/plugins/data/public/antlr/opensearch_sql/code_completion.ts

```diff
@@ -47,80 +48,59 @@ export const getSuggestions = async ({
-          const fieldTypes = schema.body.fields.find((col: any) => col.name === 'DATA_TYPE');
+    if (suggestions.suggestColumns?.tables?.length) {
+      const tableNames = suggestions.suggestColumns.tables.map((table) => table.name);
+      const schemas = await fetchTableSchemas(tableNames, api, dataSetManager);
+
+      (schemas as IDataFrameResponse[]).forEach((schema: IDataFrameResponse) => {
+        if ('body' in schema && schema.body && 'fields' in schema.body) {
+          const columns = schema.body.fields.find((col: IFieldType) => col.name === 'COLUMN_NAME');
+          const fieldTypes = schema.body.fields.find((col: IFieldType) => col.name === 'DATA_TYPE');
+
           if (columns && fieldTypes) {
             finalSuggestions.push(
-              ...columns.values.map((col: string, index: number) => ({
+              ...columns.values.map((col: string) => ({
                 text: col,
-                type: 'field',
-                fieldType: fieldTypes.values[index],
+                type: monaco.languages.CompletionItemKind.Field,
               }))
             );
           }
         }
       });
-
-      if (
-        'suggestValuesForColumn' in suggestions &&
-        /\S/.test(suggestions.suggestValuesForColumn as string) &&
-        suggestions.suggestValuesForColumn !== undefined
-      ) {
-        const values = await fetchColumnValues(
-          tableNames,
-          suggestions.suggestValuesForColumn as string,
-          api,
-          services
-        );
-        values.forEach((value) => {
-          if (value.body?.fields?.length > 0) {
-            finalSuggestions.push(
-              ...value.body.fields[0].values.map((colVal: string) => ({
-                text: `'${colVal}'`,
-                type: 'value',
-              }))
-            );
-          }
-        });
-      }
     }
 
     // Fill in aggregate functions
-    if ('suggestAggregateFunctions' in suggestions && suggestions.suggestAggregateFunctions) {
+    if (suggestions.suggestAggregateFunctions) {
       finalSuggestions.push(
         ...SQL_SYMBOLS.AGREGATE_FUNCTIONS.map((af) => ({
           text: af,
-          type: 'function',
+          type: monaco.languages.CompletionItemKind.Function,
         }))
       );
     }
 
     // Fill in SQL keywords
-    if ('suggestKeywords' in suggestions && (suggestions.suggestKeywords?.length ?? 0) > 0) {
+    if (suggestions.suggestKeywords?.length) {
       finalSuggestions.push(
-        ...(suggestions.suggestKeywords ?? []).map((sk) => ({
+        ...suggestions.suggestKeywords.map((sk) => ({
           text: sk.value,
-          type: 'keyword',
+          type: monaco.languages.CompletionItemKind.Keyword,
         }))
       );
     }
   } catch (error) {
-    // TODO: pipe error to the UI
+    // TODO: Handle errors appropriately, possibly logging or displaying a message to the user
+    return [];
```

Comment:
is this catching errors for `await fetchTableSchemas` or something else could also throw error? if the network call fails, do we want to show other suggestions like `suggestAggregateFunctions` if possible?

Reply:
Yea we could but the only concern here is if fetching the schema is having error, but user is still getting other suggestions, would this error being potentially hidden in this case that user still thinks everything is working accurately?

Reply:
i would think that there are different completion sources, one source failing shouldn't affect others. the TODO `Handle errors appropriately, possibly logging or displaying a message to the user` should help to notify user about any failure